rc/extra/gdb.kak

```diff
@@ -0,0 +1,282 @@
+        fi
+    }
+}
+
+define-command gdb-run              %{ gdb-cmd run }
+define-command gdb-start            %{ gdb-cmd start }
+define-command gdb-step             %{ gdb-cmd step }
+define-command gdb-next             %{ gdb-cmd next }
+define-command gdb-finish           %{ gdb-cmd finish }
+define-command gdb-continue         %{ gdb-cmd continue }
+define-command gdb-advance          %{ gdb-cmd "advance %val{buffile}:%val{cursor_line}" }
+define-command gdb-set-breakpoint   %{ gdb-cmd "break %val{buffile}:%val{cursor_line}" }
+define-command gdb-clear-breakpoint %{ gdb-cmd "clear %val{buffile}:%val{cursor_line}" }
+define-command gdb-toggle-breakpoint %{
+    try %{
+        %sh{
+            printf %s\\n "$kak_opt_gdb_breakpoints_info" | tr ':' '\n' |
+            while read -r current; do
+                buffer=$(printf %s "$current" | cut -d \| -f 4-)
+                line=$(printf %s "$current" | cut -d \| -f 3)
+                if [ "$buffer" = "$kak_buffile" -a "$line" = "$kak_cursor_line" ]; then
+                    printf %s\\n raise
+                fi
+            done
+        }
+        gdb-set-breakpoint
+    } catch %{
+        gdb-clear-breakpoint
+    }
+}
+define-command gdb-print %{ gdb-cmd "print %val{selection}" }
+
+# implementation details commands
+
+define-command -hidden -params 2 gdb-handle-stopped %{
+    set-option global gdb_location_info "%arg{1}|%arg{2}"
+    gdb-refresh-location-flag
+}
+
+define-command -hidden gdb-refresh-location-flag %{
+    try %{
+        %sh{
+            if [ -n "$kak_opt_gdb_location_info" ]; then
+                line=$(printf %s "$kak_opt_gdb_location_info" | cut -d \| -f 1)
+                buffer=$(printf %s "$kak_opt_gdb_location_info" | cut -d \| -f 2-)
+                printf "set-option -add \"buffer=%s\" gdb_location_flag \"%s|->\"\n" "$buffer" "$line"
+            fi
+        }
+    }
+}
+
+define-command -hidden gdb-clear-location %{
+    %sh{
+        if [ -n "$kak_opt_gdb_location_info" ]; then
+            buffer=$(printf %s "$kak_opt_gdb_location_info" | cut -d \| -f 2-)
+            printf "unset-option \"buffer=%s\" gdb_location_flag" "$buffer"
+        fi
+    }
+    set-option global gdb_location_info ""
+}
+
+define-command -hidden -params 4 gdb-handle-breakpoint-created %{
+    set-option -add global gdb_breakpoints_info "%arg{1}|%arg{2}|%arg{3}|%arg{4}"
+    gdb-refresh-breakpoints-flags %arg{4}
+}
+
+define-command -hidden -params 1 gdb-handle-breakpoint-deleted %{
+    %sh{
+        printf "set-option global gdb_breakpoints_info \"\"\n"
+        while read -r current; do
+            id=$(printf %s "$current" | cut -d \| -f 1)
+            if [ "$id" != "$1" ]; then
+                printf "set-option -add global gdb_breakpoints_info \"%s\"\n" "$current"
+            else
+                buffer=$(printf %s "$current" | cut -d \| -f 4-)
+            fi
+        done <<-EOF
+			$(printf %s "$kak_opt_gdb_breakpoints_info" | tr ':' '\n')
+		EOF
```

Comment:
Is there something amiss here? There's unexpected indentation and an here-document delimiter.